staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/factory.go

```diff
@@ -61,3 +75,17 @@ func CreateReadyCheck(c storagebackend.Config, stopCh <-chan struct{}) (func() e
 		return nil, fmt.Errorf("unknown storage type: %s", c.Type)
 	}
 }
+
+// Create creates a storage backend based on given config.
+func Create(c storagebackend.ConfigForResource, newFunc func() runtime.Object) (storage.Interface, DestroyFunc, error) {
+	return DefaultStorage.Create(c, newFunc)
+}
```

Comment:
trying to understand how this is called

it seems it comes from 

https://github.com/kubernetes/kubernetes/blob/2a88664eccdf02cd01777b48171873a028117737/staging/src/k8s.io/apiserver/pkg/registry/generic/storage_decorator.go#L56-L58

and this from 

https://github.com/kubernetes/kubernetes/blob/610b7839a0bc2f9bafaca92930f8dd740f0ab46c/staging/src/k8s.io/apiserver/pkg/server/options/etcd.go#L267

and from

https://github.com/kubernetes/kubernetes/blob/d046a58de435a07cf5d0d2d0879c3b21dbb52cca/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1375
https://github.com/kubernetes/kubernetes/blob/d046a58de435a07cf5d0d2d0879c3b21dbb52cca/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L1196

Reply:
I took a pass at following through this at https://github.com/kubernetes/kubernetes/issues/111476#issuecomment-1198803133 - the decorator bit was a little unintuitive.